### ISSUE TITLE
Jetpack.com: keyboard users can't close language selector modal

### ISSUE THREAD

**Issue by jkguidaven** [Automattic/jetpack#30177]
## What's the issue
The language selector modal in Jetpack.com can't be closed by keyboard users.

## How to reproduce the issue
- Visit https://jetpack.com/
- Scroll to the footer
- Open the language selector modal
- Use the `Tab` key to navigate through interactive elements. Note that you can't close the modal.

## Expected behavior

- The content of the page behind the modal shouldn't be able to receive focus
- The close button should be focusable (and focus state should be visible)
- Users should be able to close the modal by pressing the `Esc` key

## Captures

![Image](https://user-images.githubusercontent.com/1620183/229593048-ca02a5d2-451a-4ac9-9952-8d3b22659366.png)

**Comment by jkguidaven** [comment]
diffs:
D108256-code
D108261-code

**Comment by jkguidaven** [comment]
Both diffs had been deployed
